docs/how-to-build-a-lexical-model.md

```diff
@@ -0,0 +1,365 @@
+
+Now move the wordlist to the appropriate place:
+
+    mv /type/the/current/path/to/wordlist.tsv example/str.tutorial/source/
+
+I now have the following folder structure:
+
+```
+.
+└── example
+    └── str.tutorial
+        └── source
+            └── wordlist.tsv
+```
+
+We need to create the remaining files:
+ - The "model info" file: `example.str.tutorial.model_info`
+ - The package file: `example.str.tutorial.model.kps`
+ - The model configuration file: `model.ts`
+
+
+Step 4.1: Creating the model info file
+--------------------------------------
+
+The model info contains a little bit of metadata that is required when
+distributing packages. It is a JSON file that lists the model license
+and a short description of the lexical model.
+
+The `.model_info` file should follow this template:
+
+```json
+{
+    "license": "<SPDX license identifier>",
+    "description": "<short description of the lexical model>"
+}
+```
+
+Copy this template, and use it to create your own `.model_info` file.
+
+Save this file to `{author}/{bcp47}.{uniq}/{author}.{bcp47}.{uniq}.model_info`.
+
+> NOTE: the only current valid value for `license` is `"mit"`.
+
+In our `example.str.tutorial` example, I'll use the following
+`.model_info`.
+
+```json
+{
+    "license": "mit",
+    "description": "Predicts words in the SENĆOŦEN language (Saanich Dialect)"
+}
+```
+
+I'll save this file to `example/str.tutorial/example.str.tutorial.model_info`.
+
+
+Step 4.2: Creating the KPS file
+-------------------------------
+
+**TODO:**
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<Package>
+  <System>
+    <KeymanDeveloperVersion>12.0.1500.0</KeymanDeveloperVersion>
+    <FileVersion>12.0</FileVersion>
+  </System>
+  <Options>
+    <FollowKeyboardVersion/>
+  </Options>
+  <Info>
+    <Name URL="">SENĆOŦEN (Saanich Dialect) Lexical Model</Name>
+    <Copyright URL="">© 2019 National Research Council Canada</Copyright>
+    <Author URL="mailto:Eddie.Santos@nrc-cnrc.gc.ca">Eddie Antonio Santos</Author>
+    <Version>1.0.3</Version>
+  </Info>
+  <Files>
+    <File>
+      <Name>..\build\nrc.str.sencoten.model.js</Name>
+      <Description>Lexical model nrc.str.sencoten.model.js</Description>
+      <CopyLocation>0</CopyLocation>
+      <FileType>.model.js</FileType>
+    </File>
+  </Files>
+  <LexicalModels>
+    <LexicalModel>
+      <Name>SENĆOŦEN dictionary</Name>
+      <ID>nrc.str.sencoten</ID>
+      <Version>1.0.3</Version>
+      <Languages>
+        <Language ID="str">North Straits Salish</Language>
+        <Language ID="str-Latn">SENĆOŦEN</Language>
+      </Languages>
+    </LexicalModel>
+  </LexicalModels>
+</Package>
+```
+
+Step 4.3: Creating the model file
+---------------------------------
+
+Finally, the `model.ts` is a TypeScript source code file that is
+executed to build the model. For our purposes, we can simply copy-paste
+the following example:
+
+```typescript
+import LexicalModelCompiler from "@keymanapp/developer-lexical-model-compiler";
```

Comment:
Also note: when we integrate as part of Keyman Developer, we'll be making this compiler available as part of the install process.

Reply:
Could you please elaborate on this point?

Reply:
Once we have Keyman Developer capable of doing lexical model builds, there won't be a need to hand-write all parts of these files. It will include a template for a model file to start people off and will make sure the compiler is readily available without needing to clone the repo.

I guess the comment was more a reflection on the current reality that that import statement requires a clone of the keymanapp/keyman repo which is somewhat less than ideal!